pallets/stateful-storage/src/lib.rs

```diff
@@ -470,15 +506,176 @@ pub mod pallet {
+			#[pallet::compact] intent_id: IntentId,
+			#[pallet::compact] target_hash: PageHash,
+			actions: BoundedVec<
+				crate::types::ItemActionV2<T::MaxItemizedBlobSizeBytes>,
+				T::MaxItemizedActionsCount,
+			>,
+		) -> DispatchResult {
+			let key = ensure_signed(origin)?;
+			let is_pruning = actions.iter().any(|a| matches!(a, ItemActionV2::Delete { .. }));
+			let caller_msa_id = Self::check_msa_and_grants(key, state_owner_msa_id, intent_id)?;
+			let caller_is_state_owner = caller_msa_id == state_owner_msa_id;
+			Self::check_intent_for_write(
+				intent_id,
+				PayloadLocation::Itemized,
+				caller_is_state_owner,
+				is_pruning,
+			)?;
+			Self::update_itemized(state_owner_msa_id, intent_id, target_hash, actions)?;
+			Ok(())
+		}
+
+		/// Deletes a Paginated storage
+		///
+		/// Note: if called by the state owner, call may succeed even on `SignatureRequired` schemas.
+		/// The fact that the entire (signed) transaction is submitted by the owner's keypair is
+		/// considered equivalent to supplying a separate signature. Note in that case that a delegate
+		/// submitting this extrinsic on behalf of a user would fail.
+		///
+		/// # Events
+		/// * [`Event::PaginatedPageDeleted`]
+		///
+		#[pallet::call_index(10)]
+		#[pallet::weight(T::WeightInfo::delete_page())]
+		pub fn delete_page_v2(
+			origin: OriginFor<T>,
+			#[pallet::compact] state_owner_msa_id: MessageSourceId,
+			#[pallet::compact] intent_id: IntentId,
+			#[pallet::compact] page_id: PageId,
+			#[pallet::compact] target_hash: PageHash,
+		) -> DispatchResult {
+			let provider_key = ensure_signed(origin)?;
+			ensure!(page_id <= T::MaxPaginatedPageId::get(), Error::<T>::PageIdExceedsMaxAllowed);
+			let caller_msa_id =
+				Self::check_msa_and_grants(provider_key, state_owner_msa_id, intent_id)?;
+			let caller_is_state_owner = caller_msa_id == state_owner_msa_id;
+			Self::check_intent_for_write(
+				intent_id,
+				PayloadLocation::Paginated,
+				caller_is_state_owner,
+				true,
+			)?;
+			Self::delete_paginated(state_owner_msa_id, intent_id, page_id, target_hash)?;
+			Ok(())
+		}
+
+		/// Applies the Add or Delete Actions on the requested Itemized page that requires signature.
+		/// Since the signature of delegator is checked, there is no need for delegation validation.
+		/// This is treated as a transaction so either all actions succeed or none will be executed.
+		///
+		/// # Events
+		/// * [`Event::ItemizedPageUpdated`]
+		/// * [`Event::ItemizedPageDeleted`]
+		///
+		#[pallet::call_index(11)]
+		#[pallet::weight(
+		T::WeightInfo::apply_item_actions_with_signature_v2_delete(payload.actions.len() as u32)
+		.max(T::WeightInfo::apply_item_actions_with_signature_v2_add(Pallet::<T>::sum_add_actions_bytes_v2(&payload.actions)))
+        )]
+		pub fn apply_item_actions_with_signature_v3(
+			origin: OriginFor<T>,
+			delegator_key: T::AccountId,
+			proof: MultiSignature,
+			payload: crate::types::ItemizedSignaturePayloadV3<T>,
```

Comment:
nit, might need to import

```suggestion
			payload: ItemizedSignaturePayloadV3<T>,
```